client/blocks/reader-post-card/index.jsx

```diff
@@ -115,97 +112,70 @@ export default class ReaderPostCard extends React.Component {
 			onCommentClick,
 			showPrimaryFollowButton,
 			isSelected,
-			showEntireExcerpt,
-			useBetterExcerpt,
 			showSiteName,
 			followSource,
 		} = this.props;
-		const isPhotoOnly = !! ( post.display_type & DisplayTypes.PHOTO_ONLY );
-		const isGallery = !! ( post.display_type & DisplayTypes.GALLERY );
+
+		const isPhotoPost = !! ( post.display_type & DisplayTypes.PHOTO_ONLY );
+		const isGalleryPost = !! ( post.display_type & DisplayTypes.GALLERY );
+		const isDiscoverPost = isDiscover( post );
+
 		const classes = classnames( 'reader-post-card', {
 			'has-thumbnail': !! post.canonical_media,
-			'is-photo': isPhotoOnly,
-			'is-gallery': isGallery,
+			'is-photo': isPhotoPost,
+			'is-gallery': isGalleryPost,
 			'is-selected': isSelected,
-			'is-showing-entire-excerpt': showEntireExcerpt
-		} );
-		const showExcerpt = ! isPhotoOnly;
-		const excerptAttribute = useBetterExcerpt && trim( post.better_excerpt ) ? 'better_excerpt' : 'excerpt';
-		let title = truncate( post.title, {
-			length: 140,
-			separator: /,? +/
+			'is-discover': isDiscoverPost
 		} );
-		const isDiscoverPost = DiscoverHelper.isDiscoverPost( post );
-		const discoverBlogName = isDiscoverPost && get( post, 'discover_metadata.attribution.blog_name' );
 
-		if ( ! title && isPhotoOnly ) {
-			title = '\xa0'; // force to non-breaking space if empty so that the title h1 doesn't collapse and complicate things
-		}
-
-		let followUrl;
-		if ( showPrimaryFollowButton ) {
-			if ( isDiscoverPost ) {
-				followUrl = DiscoverHelper.getSourceFollowUrl( post );
-			} else {
-				followUrl = feed ? feed.feed_URL : post.site_URL;
-			}
-		}
-
-		let featuredAsset;
-		if ( ! post.canonical_media ) {
-			featuredAsset = null;
-		} else if ( post.canonical_media.mediaType === 'video' ) {
-			featuredAsset = <FeaturedVideo { ...post.canonical_media } videoEmbed={ post.canonical_media } />;
+		const title = truncate( post.title, { length: 140, separator: /,? +/ } );
+
+		const discoverBlogName = getDiscoverBlogName( post );
```

Comment:
Shall we only do this if we know it's a Discover post, to safe the extra work for most posts?